src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs

```diff
@@ -425,6 +425,15 @@ internal static void DecodeMethodImplAttribute<T, TAttributeSyntaxNode, TAttribu
                     messageProvider.ReportInvalidAttributeArgument(arguments.Diagnostics, arguments.AttributeSyntaxOpt, 0, attribute);
                     options = options & ~(MethodImplOptions)3;
                 }
+
+                // https://github.com/dotnet/roslyn/issues/79792: Use the real value when possible
+                const MethodImplOptions MethodImplOptionsAsync = (MethodImplOptions)0x2000;
+                if ((options & MethodImplOptionsAsync) != 0)
```

Comment:
nit: consider a test with a combination of values (ie. demonstrates that we're checking the flag, not the entire value)